src/LiveComponent/src/Attribute/LiveProp.php

```diff
@@ -99,14 +101,19 @@ public function __construct(
         private null|string|array $onUpdated = null,
 
         /**
-         * If true, this property will be synchronized with a query parameter
-         * in the URL.
+         * Whether to synchronize this property with a query parameter
+         * in the URL. Pass true to configure the mapping automatically, or a
+         * {@see QueryMapping} instance to configure the mapping.
          */
-        private bool $url = false,
+        private bool|QueryMapping $url = false,
```

Comment:
Instead of having the mapping in the url argument, is it not more readable to have a separate argument alias?